Automattic/wp-calypso
# GS on personal A/B: Tootips to reflect the A/B plan.

Related to https://github.com/Automattic/dotcom-forge/issues/2859

## Proposed Changes

This PR adjusts the message displayed in the `ThemeTypeBadgeTooltip` component based on a site's global styles status. The tooltip message differentiates between sites that have global styles in their Personal plan, and those that require a Premium plan upgrade to unlock this feature. We achieve this by utilizing the `useSiteGlobalStylesStatus` hook to fetch the site's global styles status and then adjusting the tooltip message accordingly.

The scope of these changes is limited to A/B testing GS on personal.

| Before | After |
------|------
![Captura de Pantalla 2023-07-04 a las 11 17 05](https://github.com/Automattic/wp-calypso/assets/1989914/d1439432-28cc-49b1-bf57-9fa6c1939fa2)|![Captura de Pantalla 2023-07-04 a las 11 18 01](https://github.com/Automattic/wp-calypso/assets/1989914/13790e19-817f-49e6-a26d-e897021c1ffb)

## Testing Instructions



1. Open the Calypso live link.
2. Make sure your site is on a `FREE` plan.
3. Navigate to the themes section either in the administration section `/themes/{yoursite}` or the site creation flow `/setup/site-setup/designSetup?siteSlug={yoursite}&ref=site-selector`
4. Select a style variation from a free theme that has style variations
5. If your site has global styles included in the Personal plan, the tooltip should read, "Unlock this style, and tons of other features, by upgrading to a Personal plan."
6. If your site doesn't have global styles included in the Personal plan, the tooltip should read, "Unlock this style, and tons of other features, by upgrading to a Premium plan."

You can change your group here (21268-explat-experiment), you also need to clear the cache when switching between groups by executing this command in your sandbox:
`wp_cache_delete( 'global-styles-on-personal-{siteId}', 'a8c_experiments' );` 

## Pre-merge Checklist
